.github/pull_request_template.md

```diff
@@ -0,0 +1,12 @@
+<!--  Thanks for sending a pull request!  Here are some tips for you:
+1. Make sure to read the Contributing Guide before submitting your PR: https://github.com/caikit/caikit/blob/main/CONTRIBUTING.md
+2. If this PR closes another issue, add 'closes #<issue number>' somewhere in the PR summary. GitHub will automatically close that issue when this PR gets merged. Alternatively, adding 'refs #<issue number>' will not close the issue, but help provide the reviewer more context.-->
+
+**What this PR does / why we need it**:
```

Comment:
Bit of a nitpick, but I usually prefer using `##` for section headers so that you get anchor links. Not really needed for small prs, but can be helpful when the body gets big